tests/src/core/geometry/testqgscircularstring.cpp

```diff
@@ -1623,6 +1625,48 @@ void TestQgsCircularString::sumUpArea()
   QGSCOMPARENEAR( area, 12.566370614359172, 0.0001 );
 }
 
+void TestQgsCircularString::sumUpArea3D()
+{
+  QgsCircularString cs;
+  double area3D = 1.0; // sumUpArea3D adds to area, so start with non-zero value
+
```

Comment:
the below results are not obvious to me (not saying they are wrong, but cannot be easily verified immediately).
I would suggest doing an half circle like (0,0),(1 1),(2 0), and rotate it so it fits on the plane of x = 0 and y = 0  (i.e.  (0 0 0),(0 1 1),(0 2 0)   and (0 0 0),(1 0 1),(2 0 2)   and their symetric), so they have all the same areas (with +/- signs)